src/plugins/autofill/autofill.js

```diff
@@ -231,47 +234,80 @@ export class Autofill extends BasePlugin {
 
     if (startOfDragCoords && startOfDragCoords.row > -1 && startOfDragCoords.col > -1) {
       const selectionData = this.getSelectionData();
-      const beforeAutofillHook = this.hot.runHooks('beforeAutofill', startOfDragCoords, endOfDragCoords, selectionData);
 
-      if (beforeAutofillHook === false) {
+      const sourceRange = {
+        from: {
+          row: Math.min(selectionRangeLast.from.row, selectionRangeLast.to.row),
+          col: Math.min(selectionRangeLast.from.col, selectionRangeLast.to.col)
+        },
+        to: {
+          row: Math.max(selectionRangeLast.from.row, selectionRangeLast.to.row),
+          col: Math.max(selectionRangeLast.from.col, selectionRangeLast.to.col)
+        }
+      };
```

Comment:
While working with the integration I noticed this line. IMO the `sourceRange` and `targetRange` coords should use the `CellRange` instance as a coordinates representation. This is not a principle spoken aloud but it is good practice to use always the same pattern within the whole project.

Moreover, the `CellRange` (or `CellCoords`) gives you some built-in methods that you can use while processing the coords. 

```suggestion
      cconst sourceRange = selectionRangeLast.clone();
```

After the changes the formulas' autofill hooks module could be simplified by using the built-in methods e.g. 
https://github.com/handsontable/handsontable/blob/59e6f74e86c59def1d5714572639742616dde389/src/plugins/formulas/autofill.js#L46-L54
to
```js
    const sourceSize = {
      width: source.getWidth(),
      height: source.getHeight()
    };

    const targetSize = {
      width: target.getWidth(),
      height: target.getHeight()
    };
``` 
and probably more.

Reply:
Discussion continues at https://github.com/handsontable/handsontable/pull/7648#discussion_r637600959